From be46c7b596dcf525a448f31b95b0f6f0d0e3e006 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Tue, 19 Mar 2024 16:53:27 +0200
Subject: [PATCH] [SW-137122] gaudi3: erom rec: fix agent code running slowness

when we do fwupd in eROM, it is loading and jumped to agent code.
Here agent code is running slow, HOST and agent code communication
protocol is going out of sync and agent couldn't update the fw.

So added c2c pll initialization to overcome the slowness.

Change-Id: I397564656c8e89d55378e80b3db5818c7bea307a
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/arch/gaudi3/common/asic.c b/arch/gaudi3/common/asic.c
index 3fa3dcc..985f5b6 100644
--- a/arch/gaudi3/common/asic.c
+++ b/arch/gaudi3/common/asic.c
@@ -58,6 +58,26 @@
 				 HL_PLL_DIV_PLL,
 				 HL_PLL_DIV_PLL
 		},
+	}, {
+		.name = "G1C2C",
+		.base = mmD0_G1_C2C_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "G5C2C",
+		.base = mmD0_G5_C2C_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL
+		},
 	}
 };
 
